runtime/connectors/blob/parquetreader.go

```diff
@@ -0,0 +1,217 @@
+package blob
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sync/atomic"
+
+	"github.com/apache/arrow/go/v11/arrow"
+	"github.com/apache/arrow/go/v11/arrow/array"
+	"github.com/apache/arrow/go/v11/arrow/memory"
+	"github.com/apache/arrow/go/v11/parquet"
+	"github.com/apache/arrow/go/v11/parquet/compress"
+	"github.com/apache/arrow/go/v11/parquet/file"
+	"github.com/apache/arrow/go/v11/parquet/pqarrow"
+	"github.com/c2h5oh/datasize"
+	"gocloud.dev/blob"
+)
+
+var batchSize = int64(1000)
```

Comment:
1. Should be `const`
2. Rename to provide more context (is it listings, file downloads, file contents? rows, bytes?)